packages/react/src/components/TwoColumnPanel/TwoColumnPanel.tsx

```diff
@@ -0,0 +1,244 @@
+        columnLeft.props.id || useId(undefined, 'sidebar-'));
+      const CloseButton = (
+        <div className="TwoColumnPanel__Close">
+          <button
+            type="button"
+            onClick={togglePanel}
+            ref={closeButtonRef}
+            aria-label={hideCollapsedPanelLabel}
+          >
+            <Icon type="close" />
+          </button>
+          <Tooltip
+            target={closeButtonRef}
+            association="aria-labelledby"
+            hideElementOnHidden
+          >
+            {hideCollapsedPanelLabel}
+          </Tooltip>
+        </div>
+      );
+      const children = [
+        CloseButton,
+        ...React.Children.toArray(columnLeft.props.children)
+      ];
+      ColumnLeftComponent = cloneElement(
+        columnLeft,
+        { id, ref, tabIndex: -1 },
+        children
+      );
+    }
+
+    const columnRight = React.Children.toArray(children).find(
+      child => (child as React.ReactElement<any>).type === ColumnRight
+    );
+
+    let ColumnRightComponent;
+    if (isValidElement(columnRight)) {
+      const ref = columnRight.props.ref || columnRightRef;
+      const ToggleButton = (
+        <div className="TwoColumnPanel__ButtonToggle">
+          <button
+            type="button"
+            onClick={togglePanel}
+            ref={toggleButtonRef}
+            aria-label={
+              !isCollapsed ? hideCollapsedPanelLabel : showCollapsedPanelLabel
+            }
+            aria-expanded={!isCollapsed}
+            aria-controls={columnLeftId}
+          >
+            <Icon
+              type={
+                !isCollapsed ? 'chevron-double-left' : 'chevron-double-right'
+              }
+            />
+          </button>
+          <Tooltip
+            target={toggleButtonRef}
+            association="aria-labelledby"
+            hideElementOnHidden
+          >
+            {!isCollapsed ? hideCollapsedPanelLabel : showCollapsedPanelLabel}
+          </Tooltip>
+        </div>
+      );
+      const children = [
+        ToggleButton,
+        ...React.Children.toArray(columnRight.props.children)
+      ];
+      ColumnRightComponent = cloneElement(
+        columnRight,
+        { ref, tabIndex: -1 },
+        children
+      );
+    }
+
+    useLayoutEffect(() => {
+      const handleTransitionEnd = () => {
+        if (columnLeftRef.current && isCollapsed) {
+          setShowPanel(false);
+        }
+      };
+
+      columnLeftRef.current?.addEventListener(
+        'transitionend',
+        handleTransitionEnd
+      );
+
+      return () => {
+        columnLeftRef.current?.removeEventListener(
+          'transitionend',
+          handleTransitionEnd
+        );
+      };
+    }, [columnLeftRef.current, isCollapsed]);
+
+    // When the collapsable panel starts to overlay content, it needs to become a focus trap and collapsed by default
+    useLayoutEffect(() => {
+      const mediaQueryList = matchMedia('(max-width: 45rem)');
```

Comment:
Two suggestions:

1. Should this media query be configurable?
2. Should we go ahead and create a `useMediaQuery` utility in cauldron to make this more reusable?